src/nimony/expreval.nim

```diff
@@ -506,6 +623,19 @@ proc eval*(c: var EvalContext; n: var Cursor): Cursor =
         evalOrdBinOp(c, n, `div`)
     of ModX:
       evalOrdBinOp(c, n, `mod`)
+    of BitorX:
+      evalOrdBinOp(c, n, `or`)
+    of BitandX:
+      evalOrdBinOp(c, n, `and`)
+    of BitxorX:
+      evalOrdBinOp(c, n, `xor`)
+    of BitnotX:
+      evalBitnot(c, n)
+    of ShlX:
+      evalShiftOp(c, n, `shl`)
+    of ShrX, AshrX:
+      # considered the same for now
```

Comment:
Nope, don't offer `ashr` until it's correct. "cannot eval ashr at compile-time" is **much** better than having to hunt subtle bugs.

Reply:
The problem is that the `shr` overloads for integers use `AshrI` when the unsigned integers use `ShrI`, and `ashr` is only defined for signed integers. I will make it just only allow the respective type for each operation